### PR TITLE
[FLINK-7822][FLINK-7823] Adds documentation and fixes configuration of QS.

### PR DESCRIPTION
**Pull request by kl0u**
## What is the purpose of the change

*This PR adds documentation for the new queryable state.

R @aljoscha

### CODE REVIEW

File: docs/dev/stream/state/queryable_state.md [hunk 1]
```diff
@@ -32,38 +32,67 @@ under the License.
...
+  <i>e.g.</i> <code>MemoryStateBackend</code> or <code>FsStateBackend</code>, does not work 
+  with copies when retrieving values but instead directly references the stored values, read-modify-write 
+  patterns are unsafe and may cause the queryable state server to fail due to concurrent modifications.
+  The <code>RocksDBStateBackend</code> is safe from these issues.
 </div>
 
+## Architecture
+
+Before showing how to use the Queryable State, it is useful to briefly describe the entities that compose it.
+The Queryable State consists of three main entities:
```
Comment (aljoscha) [inline docs/dev/stream/state/queryable_state.md]: nit: "The Queryable State feature"

File: docs/dev/stream/state/queryable_state.md [hunk 2]
```diff
@@ -32,38 +32,67 @@ under the License.
...
+
+ 1. the `QueryableStateClient`, which (potentially) runs outside the Flink cluster and submits the user queries, 
+ 2. the `QueryableStateClientProxy`, which runs on each `TaskManager` (*i.e.* inside the Flink cluster) and is responsible 
+ for receiving the client's queries, fetching the requested state on his behalf, and returning it to the client, and
```
Comment (aljoscha) [inline docs/dev/stream/state/queryable_state.md]: maybe "fetching the requested state from the responsible TaskManager"

File: docs/dev/stream/state/queryable_state.md [hunk 3]
```diff
@@ -32,38 +32,67 @@ under the License.
...
+In a nutshell, the client will connect to one of the proxies and send a request for the state associated with a specific 
+key, `k`. As stated in [Working with State]({{ site.baseurl }}/dev/stream/state/state.html), keyed state is organized in 
+*Key Groups*, and each `TaskManager` is assigned a number of these key groups. To discover which `TaskManager` is 
+responsible for the key group holding `k`, the proxy will ask the `JobManager`. Based on the answer, the proxy will 
+then query the `QueryableStateServer` running on that `TaskManager` for the state associated with `k`, and forward the
+response back to the client.
+
+## Activating Queryable State
+
+To enable queryable state on your Flink cluster, you just have to copy the
```
Comment (aljoscha) [inline docs/dev/stream/state/queryable_state.md]: nit: omit "the"

File: docs/dev/stream/state/queryable_state.md [hunk 4]
```diff
@@ -32,38 +32,67 @@ under the License.
...
+responsible for the key group holding `k`, the proxy will ask the `JobManager`. Based on the answer, the proxy will 
+then query the `QueryableStateServer` running on that `TaskManager` for the state associated with `k`, and forward the
+response back to the client.
+
+## Activating Queryable State
+
+To enable queryable state on your Flink cluster, you just have to copy the 
+`flink-queryable-state-runtime{{ site.scala_version_suffix }}-{{site.version }}.jar` 
+from the `opt/` folder of your [Flink distribution](https://flink.apache.org/downloads.html "Apache Flink: Downloads"), 
+to the `lib/` folder. In other case, the queryable state feature is not enabled.
```
Comment (aljoscha) [inline docs/dev/stream/state/queryable_state.md]: nit: "Otherwise, ..."

File: docs/dev/stream/state/queryable_state.md [hunk 5]
```diff
@@ -32,38 +32,67 @@ under the License.
...
+
+To verify that your cluster is running with queryable state enabled, check the logs of any 
+task manager for the line: `"Started the Queryable State Proxy Server @ ..."`.
+
 ## Making State Queryable
 
-In order to make state queryable, the queryable state server first needs to be enabled globally
-by setting the `query.server.enable` configuration parameter to `true` (current default).
-Then appropriate state needs to be made queryable by using either
+Now that you have activated queryable state on your cluster, it is time to see how to use it. In order for a state to
```
Comment (aljoscha) [inline docs/dev/stream/state/queryable_state.md]: "state to be visible"

File: docs/dev/stream/state/queryable_state.md (hunk 3 above)
Comment (kl0u) [reply docs/dev/stream/state/queryable_state.md]: You mean make it: "To enable **the** queryable state on your Flink cluster..."?

**Pull request merged**